wemake_python_styleguide/visitors/ast/naming/variables.py

```diff
@@ -86,11 +86,14 @@ def _check_reassignment(
 
         for var_name, var_value in itertools.zip_longest(names, var_values):
             if var_name == var_value:
-                self.add_violation(
-                    best_practices.ReassigningVariableToItselfViolation(
-                        node, text=var_name,
-                    ),
-                )
+                if type(var_name)==bool or type(var_value) == bool:
```

Comment:
Checking types should be done with identity instead of equality: type(var_name) is bool.